core/services/offchainreporting/data_source.go

```diff
@@ -2,64 +2,52 @@ package offchainreporting
 
 import (
 	"context"
-	"time"
 
+	"github.com/jinzhu/gorm"
 	"github.com/pkg/errors"
-	"github.com/smartcontractkit/chainlink/core/logger"
 	"github.com/smartcontractkit/chainlink/core/services/pipeline"
 	"github.com/smartcontractkit/chainlink/core/utils"
 	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting/types"
 )
 
 // dataSource is an abstraction over the process of initiating a pipeline run
-// and capturing the result.  Additionally, it converts the result to an
+// and capturing the result. Additionally, it converts the result to an
 // ocrtypes.Observation (*big.Int), as expected by the offchain reporting library.
 type dataSource struct {
 	pipelineRunner pipeline.Runner
 	jobID          int32
+	spec           pipeline.Spec
 }
 
 var _ ocrtypes.DataSource = (*dataSource)(nil)
 
+func newDatasource(db *gorm.DB, jobID int32, pipelineRunner pipeline.Runner) (*dataSource, error) {
+	var spec pipeline.Spec
```

Comment:
We need to grab the job.SpecDB associated with this jobID then preload PipelineTaskSpecs and PipelineSpec. I'm pretty sure this just gets the first pipeline_spec in the db?